interface/presenter/pokemon_presenter_test.go

```diff
@@ -0,0 +1,120 @@
+package presenter
+
+import (
+	"encoding/json"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/bxcodec/faker/v3"
+	"github.com/wmaldonadoc/academy-go-q42021/domain/model"
+	"github.com/wmaldonadoc/academy-go-q42021/infrastructure/api"
+	"github.com/wmaldonadoc/academy-go-q42021/interface/vendors"
+)
+
+func TestResponsePokemon(t *testing.T) {
+	tests := []struct {
+		name    string
+		pokemon *model.Pokemon
+	}{
+		{name: "Return the same pokemon passed as parameter"},
+	}
+
+	pr := NewPokemonPresenter()
+
+	for _, test := range tests {
+		id, _ := faker.RandomInt(1, 100)
+		test.pokemon = &model.Pokemon{
+			ID:      id[0],
+			Ability: faker.Word(),
+			Name:    faker.Name(),
+		}
+
+		result := pr.ResponsePokemon(test.pokemon)
+
+		if !reflect.DeepEqual(test.pokemon.ID, result.ID) {
+			t.Errorf("%s: Expected ID %d but got %d", test.name, test.pokemon.ID, result.ID)
+		}
+
+		if !reflect.DeepEqual(test.pokemon.Ability, result.Ability) {
+			t.Errorf("%s: Expected ability %s but got %s", test.name, test.pokemon.Ability, result.Ability)
+		}
+
+		if !reflect.DeepEqual(test.pokemon.Name, result.Name) {
+			t.Errorf("%s: Expected name %s but got %s", test.name, test.pokemon.Name, result.Name)
+		}
```

Comment:
You could check for the whole struct, instead of validating each single field of it.